tests/test_workers/test_pkg_managers/test_pip.py

```diff
@@ -0,0 +1,587 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import contextlib
+import logging
+from pathlib import Path
+from textwrap import dedent
+
+import pytest
+
+from cachito.workers.pkg_managers import pip
+from cachito.errors import ValidationError
+
+
+def setup_module():
+    """Re-enable logging that was disabled at some point in previous tests."""
+    pip.log.disabled = False
+    pip.log.setLevel(logging.DEBUG)
+
+
+def write_file_tree(tree_def, rooted_at):
+    """
+    Write a file tree to disk.
+
+    :param dict tree_def: Definition of file tree, see usage for intuitive examples
+    :param (str | Path) rooted_at: Root of file tree, must be an existing directory
+    """
+    root = Path(rooted_at)
+    for entry, value in tree_def.items():
+        entry_path = root / entry
+        if isinstance(value, str):
+            entry_path.write_text(value)
+        else:
+            entry_path.mkdir()
+            write_file_tree(value, entry_path)
+
+
+def remove_file_tree(tree_def, rooted_at):
+    """
+    Remove a file tree from disk.
+
+    :param dict tree_def: Definition of file tree, see usage for intuitive examples
+    :param (str | Path) rooted_at: Root of file tree, will not be removed
+    """
+    root = Path(rooted_at)
+    for entry, value in tree_def.items():
+        entry_path = root / entry
+        if isinstance(value, str):
+            entry_path.unlink()
+        else:
+            remove_file_tree(value, entry_path)
+            entry_path.rmdir()
+
+
+@contextlib.contextmanager
+def file_tree(tree_def, rooted_at):
+    """
+    Create a file tree for the duration of the context.
+
+    :param dict tree_def: Definition of file tree, see usage for intuitive examples
+    :param (str | Path) rooted_at: Root of file tree
+    """
+    write_file_tree(tree_def, rooted_at)
+    try:
+        yield
+    finally:
+        remove_file_tree(tree_def, rooted_at)
```

Comment:
Why not just `shutil.rmtree`? Or, better yet, always use `tmpdir` and let pytest clean up these files automatically.

Reply:
Good point, I had some issues with leftover files when I was trying to use `importlib.util.find_spec()` for finding modules (which also involved temporarily modifying `sys.path`). In the end, I never got it to work and the current solution no longer has this issue.